src/ui/buttons/buttonDesigns/fadeOutLogoShowLabelText.jsx

```diff
@@ -0,0 +1,121 @@
+    const { ANIMATION_CONTAINER, ANIMATION_LABEL_CONTAINER, PAYPAL_BUTTON_LABEL, PAYPAL_LOGO } = configuration.cssClasses;
+    const { tiny, medium } = configuration;
+    // get the animation main container to force specificity( in css ) and make sure we are running the right animation
+    const animationContainer = (document && document.querySelector(`.${ ANIMATION_CONTAINER }`)) || null;
+    if (!animationContainer) {
+        return null;
+    }
+
+    // return null if animation should not be played for the button size
+    const animationContainerWidth = animationContainer.offsetWidth;
+    if (animationContainerWidth < tiny.min || animationContainerWidth > medium.max) {
+        // remove label element from dom
+        animationContainer.querySelector(`.${ ANIMATION_LABEL_CONTAINER }`).remove();
+        return null;
+    }
+
+    // get the label container that animation will be applied to
+    const paypalLabelContainerElement = animationContainer.querySelector(`.${ PAYPAL_BUTTON_LABEL }`) || null;
+    // get starting position for element so it doesn't jump when animation begins
+    const paypalLogoElement = (paypalLabelContainerElement && paypalLabelContainerElement.querySelector(`.${ PAYPAL_LOGO }`)) || null;
+    const paypalLogoStartingLeftPosition = paypalLogoElement
+        ? `${ (paypalLogoElement.offsetLeft / paypalLabelContainerElement.offsetWidth) * 100 }`
+        : '44.5';
+
+    return {
+        paypalLabelContainerElement,
+        paypalLogoStartingLeftPosition
+    };
+};
+
+export function getFadeoutLogoShowLabelTextAnimation (animationProps, animationConfig) : void | null {
+    const { ANIMATION_LABEL_CONTAINER, ANIMATION_CONTAINER, DOM_READY, PAYPAL_LOGO } = animationConfig.cssClasses;
+    const { paypalLabelContainerElement, paypalLogoStartingLeftPosition } = animationProps;
+    const animations = `
+        .${ DOM_READY } .${ ANIMATION_CONTAINER } img.${ PAYPAL_LOGO }{
+            animation: 4s move-logo-to-left-side 1s infinite alternate;
+            position:fixed;
+            left: ${ paypalLogoStartingLeftPosition }%;
+        }
+        
+        .${ ANIMATION_CONTAINER } .${ ANIMATION_LABEL_CONTAINER } {
+            animation: 4s divide-logo-animation-right-side 1s infinite alternate;
+            text-align: center;
+            width: 100%;
+        }
+
+        @keyframes move-logo-to-left-side {
+            0%,33%,66% {
+                left: ${ paypalLogoStartingLeftPosition }%;
+                opacity: 1;
+            }
+            90%,100% {
+                left: 0%;
+                opacity:0;
+            }
+        }
+        
+        @keyframes divide-logo-animation-right-side {
+            0%,33%,66%{
+                opacity: 0;
+            }
+            90%, 100% {
+                opacity: 1;                    
+            }
+        }
+    `;
+
+    if (paypalLabelContainerElement) {
```

Comment:
Please add a resize listener here in case the button becomes too large for the design.
```
window.addEventListener('resize', () => {
            // Remove animation if size limit broken
            if (
                (designContainer.offsetWidth > max || designContainer.offsetWidth < min)
                && paypalLabelContainerElement.contains(style)
            ) {
                paypalLabelContainerElement.removeChild(style);
            }
        });
```

Reply:
done, btw I think you are missing another condition to enable the animation again if the user resizes between the allowed sizes, check what I did and let me know